protocol/messenger.go

```diff
@@ -2865,10 +2894,11 @@ type ReceivedMessageState struct {
 	// Response to the client
 	Response *MessengerResponse
 	// Timesource is a time source for clock values/timestamps.
-	Timesource              common.TimeSource
-	AllBookmarks            map[string]*browsers.Bookmark
-	AllVerificationRequests []*verification.Request
-	AllTrustStatus          map[string]verification.TrustStatus
+	Timesource                 common.TimeSource
+	AllBookmarks               map[string]*browsers.Bookmark
+	AllVerificationRequests    []*verification.Request
+	AllTrustStatus             map[string]verification.TrustStatus
+	SavedAddressesChangesCount int
```

Comment:
Could you elaborate what is this used for?

Reply:
Notifies clients about sync changes so they can update the saved addresses view

Reply:
I think how we generally do it is similarly to the rest of messenger response, we actually pass the new/changed objects, so we keep the behavior consistent

Reply:
Done.

> ... so we keep the behavior consistent

In desktop, we reload all the saved addresses when a change is reported hence we don't use the incremental data. Using the messenger updates by merging with the presentation layer model looked to me like a premature optimization at this point.

I updated the implementation to follow the other models and added the ChainId and Address on a delete without the rest of the data, please let me know if this will work for mobile.